/*
 * Copyright 2018- The Pixie Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

syntax = "proto3";

package experimental;

option go_package = "http";

// We can also track the removal of certain field through the proto message.
// In the process of adding a new field, let it propagate. User update the table schema to ignore
// the old field. And then remove the old field. This process is nicely captured in proto changes.
message HTTPRecord {
  uint64 time_stamp_ns = 1;  // The tag number will correspond to the index in
  // ColumnWrapperRecordBatch, off by 1.
  uint32 tgid = 2;
  uint32 fd = 3;
  string type = 4;
  string src_addr = 5;
  uint32 src_port = 6;
  string dst_addr = 7;
  uint32 dst_port = 8;
  uint32 minor_version = 9;
  string headers = 10;
  string req_method = 11;
  string req_path = 12;
  uint32 resp_status = 13;
  string resp_message = 14;
  string resp_body = 15;
  uint64 resp_latency_ns = 16;
}
